model/project.go

```diff
@@ -282,25 +289,49 @@ func (bvt *BuildVariantTaskUnit) UnmarshalYAML(unmarshal func(interface{}) error
 }
 
 func (bvt *BuildVariantTaskUnit) SkipOnRequester(requester string) bool {
+	if len(bvt.AllowedRequesters) != 0 {
+		return !utility.StringSliceContains(bvt.AllowedRequesters, requester)
+	}
+
 	return evergreen.IsPatchRequester(requester) && bvt.SkipOnPatchBuild() ||
 		!evergreen.IsPatchRequester(requester) && bvt.SkipOnNonPatchBuild() ||
 		evergreen.IsGitTagRequester(requester) && bvt.SkipOnGitTagBuild() ||
 		!evergreen.IsGitTagRequester(requester) && bvt.SkipOnNonGitTagBuild()
 }
 
 func (bvt *BuildVariantTaskUnit) SkipOnPatchBuild() bool {
+	if len(bvt.AllowedRequesters) != 0 {
+		allowed := utility.StringSliceIntersection(bvt.AllowedRequesters, evergreen.PatchRequesters)
+		return len(allowed) == 0
+	}
+
 	return !utility.FromBoolTPtr(bvt.Patchable)
 }
 
 func (bvt *BuildVariantTaskUnit) SkipOnNonPatchBuild() bool {
+	if len(bvt.AllowedRequesters) != 0 {
+		allowed, _ := utility.StringSliceSymmetricDifference(bvt.AllowedRequesters, evergreen.PatchRequesters)
+		return len(allowed) == 0
+	}
+
 	return utility.FromBoolPtr(bvt.PatchOnly)
 }
 
 func (bvt *BuildVariantTaskUnit) SkipOnGitTagBuild() bool {
+	if len(bvt.AllowedRequesters) != 0 {
+		allowed := utility.StringSliceIntersection(bvt.AllowedRequesters, []string{evergreen.GitTagRequester})
```

Comment:
nit: feels cleaner to just use stringSliceContains for git tags

Reply:
Done.